Add MessagePackWindow and some fixes

neuecc opened the pull request:
In UnityEditor, Windows -> MessagePack -> CodeGenerator shows below window.

![image](https://user-images.githubusercontent.com/46207/69414381-f14da400-0d55-11ea-9f8d-9af448d347dc.png)

* Try to check dotnet installed(if not, open dotnet core install url page)
* Try to check installed MessagePack.Generator(as .NET Core Global tools, if not call `dotnet tool install --global messagepack.generator`)
* Try to call `dotnet mpc`

---

other fixes

* remove `ENABLE_UNSAFE_MSGPACK` because v2 is always should enable  unsafe.
* rename folder `UnityShims` -> `Unity` because in Unity, it is not Shims.
* integrated to one package exporter(remove `Assets/Editor/PackageExport.cs`, should use `Assets/Scripts/Editor/PackageExporter.cs`).

AArnott commented:
Rather than install mpc as a *global* tool, how about installing it as a local tool? That way it's safer for build scripts to be written on it, it will be on every computer that needs it since the tool manifest gets checked into SCC with the project, etc. And you can still run a local tool with "dotnet mpc"

AArnott commented on src/MessagePack.UnityClient/Assets/Scripts/Editor/PackageExporter.cs:
```diff
@@ -11,14 +11,21 @@ public static void Export()
         // configure
         var root = "Scripts/MessagePack";
-        var exportPath = "./MessagePack.Unity.unitypackage"; // missing version:)
+        var exportPath = "../../bin/MessagePack.unitypackage";
```
Regarding the "missing version" comment, I noticed that recently too that in 1.x you released messagepack unitypackage files with the version built into the filename. And it wasn't obvious anywhere else what the version would be, so this seemed liked a good idea.
I don't know how/where you did that, but I can help with getting the version number to you if you can help with the part about building a file with that name.

neuecc commented:
@AArnott 
Thank you for reviewing.
I would like to make improvements after the release, MessagePackWindow.
(The configuration is currently saved in its own EditorPrefs, but it is better to read it from the project-specific settings)

> version

I've changed to `MessagePack.Unity.{version}.unitypackage` if provided commandline.
in commandline argument, `/version x.x.x` will work.
I don't know what parameter I can use in azure pipelines so please modify ci configuration?

---
I've modified 
`<AssemblyName>MessagePack.Generator</AssemblyName>`
to
`<AssemblyName>mpc</AssemblyName>`

By the way, for the first time this time, I confirmed the release asset of `v2.0.245-rc+a81989f414`.
Currently provides mpc as 
- mpc-win.zip
- mpc-osx.zip
- mpc-linux.zip

but could you provide a single zip file?
```
- mpc.zip
  - win-x64
  - osx-x64
  - linux-x64
  - any
```

AArnott commented:
I have pushed a commit that activates your change to include the version number in the unitypackage file.

Regarding the mpc.zip you requested, I can do that. But I made it three separate zip files because:
1. Each mpc executable is large: 50MB+
1. Someone who wants to use it would almost certainly want only one of them. So why make them download 3?

neuecc commented:
mpc is mainly used for Unity.
(for Xamarin users, maybe used MSTask version in v2)
In general, the game development is various environment.
For exmaple,

* Often there are both Mac developers and Windows developers.
* The local is Windows and CI is Mac (because it is essential for iOS build in Unity)
* There can also be CI on Linux (I run Unity on the CircleCI Linux container)

In other words, who going to downloading mpc, probably use multi executable.
for a casual user, I've prepare the editor window.
(To be honest, I don't want to install dotnet runtime to users, so I want to distribute SCDs well)

In v2 I've prepared many cases.
* download mpc as SCD and FDD
* install from .NET Core Global Tools
* install and execute from EditorWindow
* MSBuild Task from NuGet

If you need an executable file, download a zip, everything is there.
I think that kind of simplicity is preferable.

AArnott commented:
Alright. I'll fix that in a separate PR.
